src/sentry/users/api/serializers/user.py

```diff
@@ -396,3 +400,50 @@ def serialize(
                 "Incorrectly calling `DetailedSelfUserSerializer`. See docstring for details."
             )
         return d
+
+
+class UserSerializerWithOrgMemberships(UserSerializer):
+    def get_attrs(
+        self,
+        item_list: Sequence[User],
+        user: User | AnonymousUser | RpcUser,
+        **kwargs: Any,
+    ) -> MutableMapping[User, Any]:
+        attrs = super().get_attrs(item_list, user, **kwargs)
+
+        memberships = OrganizationMemberMapping.objects.filter(
+            user_id__in={u.id for u in item_list}
+        ).values_list("user_id", "organization_id", named=True)
+        oms = OrganizationMapping.objects.filter(
+            organization_id__in={m.organization_id for m in memberships},
+            status=OrganizationStatus.ACTIVE,
+        )
+        active_organization_ids = [om.organization_id for om in oms]
+        active_organization_names = {}
+        for om in oms:
+            active_organization_names[om.organization_id] = om.name
```

Comment:
nit: i like to use this hack to get a mapping directly from a QuerySet

```suggestion
        active_org_id_to_name = dict(OrganizationMapping.objects.filter(
            organization_id__in={m.organization_id for m in memberships},
            status=OrganizationStatus.ACTIVE,
        ).values_list("id", "name"))
        active_organization_ids = active_org_id_to_name.keys()
```

Reply:
you could also use the list of `active_organization_ids` to further filter memberships before applying the `values_list` operation

```python
memberships = OrganizationMemberMapping.objects.filter(
    user_id__in={u.id for u in item_list}
)
# get active org ids...
memberships = memberships.filter(organization_id__in=active_organization_ids).values_list("user_id", "organization_id", named=True)
```